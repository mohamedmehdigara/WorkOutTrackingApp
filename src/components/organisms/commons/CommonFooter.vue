<script setup lang="ts">
import { useRouter } from "vue-router";

const headerItems = [
  { name: "ホーム", src: "/assets/Footer/Icon_Home.png", to: "/" },
  {
    name: "スポット検索",
    src: "/assets/Footer/Icon_Search.png",
    to: "/search",
  },
  {
    name: "お気に入り",
    src: "/assets/Footer/Icon_Favorite.png",
    to: "/",
  },
  { name: "プラン", src: "/assets/Footer/Icon_Plan.png", to: "/" },
];

const router = useRouter();

const transition = (to) => {
  router.push(to);
};
</script>

<template>
  <footer
    class="h-100 flex flex-wrap flex-row bg-white fixed inset-x-0 bottom-0 pt-2"
  >
    <div
      class="w-1/4 text-center"
      v-for="item in headerItems"
      @click="transition(item.to)"
      :key="item.name"
    >
      <!-- アイコン -->
      <img :src="item.src" class="m-auto" alt="" width="20" height="17" />
      <!-- 文字 -->
      <span class="text-xxs text-black block mb-1 font-mono">
        {{ item.name }}
      </span>
    </div>
  </footer>
</template>
