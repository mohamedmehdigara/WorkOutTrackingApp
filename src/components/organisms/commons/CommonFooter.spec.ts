import { describe, it, expect } from "vitest";

import { mount } from "@vue/test-utils";
import CommonFooter from "./CommonFooter.vue";

describe("CommonFooter", () => {
  it("renders all elements properly", () => {
    const wrapper = mount(CommonFooter);
    console.log("a :", wrapper.attributes());
    expect(wrapper.html()).toContain("FOOD");
    expect(wrapper.html()).toContain("HOME");
    expect(wrapper.html()).toContain("MENU");
    expect(wrapper.html()).toContain("PROFILE");
  });
});
