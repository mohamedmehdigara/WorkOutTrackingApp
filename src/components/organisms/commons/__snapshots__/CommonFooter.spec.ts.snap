// Vitest Snapshot v1

exports[`CommonFooter > renders all elements properly 1`] = `
"<footer class=\\"mx-4 h-20 flex items-center rounded-xl mb-5 flex-wrap flex-row bg-white fixed inset-x-0 bottom-0 drop-shadow-lg border border-gray-2\\">
  <div class=\\"w-1/5 text-center\\"><img src=\\"/icons/Home.svg\\" class=\\"m-auto\\" alt=\\"\\"><span class=\\"text-xxs text-black block mb-1 font-bold\\"> HOME </span></div>
  <div class=\\"w-1/5 text-center\\"><img src=\\"/icons/Food.svg\\" class=\\"m-auto\\" alt=\\"\\"><span class=\\"text-xxs text-black block mb-1 font-bold\\"> FOOD </span></div><!-- + icon --><img src=\\"/icons/LargeAddBtn.svg\\" alt=\\"\\" class=\\"w-1/5 absolute bottom-5 left-[135px]\\">
  <div class=\\"w-1/5 text-center\\"><span class=\\"hidden\\"></span></div>
  <div class=\\"w-1/5 text-center\\"><img src=\\"/icons/Menu.svg\\" class=\\"m-auto\\" alt=\\"\\"><span class=\\"text-xxs text-black block mb-1 font-bold\\"> MENU </span></div>
  <div class=\\"w-1/5 text-center\\"><img src=\\"/icons/Profile.svg\\" class=\\"m-auto\\" alt=\\"\\"><span class=\\"text-xxs text-black block mb-1 font-bold\\"> PROFILE </span></div>
</footer>"
`;
